libs/langchain-google-common/src/connection.ts

```diff
@@ -53,11 +53,28 @@ export abstract class GoogleConnection<
 
   async _clientInfoHeaders(): Promise<Record<string, string>> {
     const clientLibraryVersion = await this._clientLibraryVersion();
+    const clientInfo= await this._getclientInfo();
     return {
       "User-Agent": clientLibraryVersion,
+      "Client-Info": clientInfo.toString(),
     };
   }
 
+  async _getclientInfo(): Promise<Record<string, string>> {
+   const userAgent = await this._clientLibraryVersion();
+   const env = await getRuntimeEnvironment();
+      const langchainVersion = env?.libraryVersion ?? "0";
+   const moduleName = await this._moduleName();
+   let clientLibraryVersion = `${langchainVersion}`;
```

Comment:
I unfortunately don't think this does what you think it does :( we don't currently have a way to extract version at runtime.